src/components/views/ViewMilestone.jsx

```diff
@@ -160,6 +163,40 @@ class ViewMilestone extends Component {
     );
   }
 
+  async shouldShowDelegateButton() {
+    const { currentUser } = this.props;
+    const { campaign } = this.state;
+    if (!campaign || !currentUser) {
```

Comment:
Some data is not loaded when the component did mount, so I cant calculate the permissions when component created, and for that particular reason we should check if all data doesn't provided we don't proceed, when the data is provided then we check the permissions